### ISSUE TITLE
geoserver - set jetty max threads parameter in docker image

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/georchestra#1666]
As seen on GeoNode doc - http://docs.geonode.org/en/master/tutorials/advanced/geonode_production/production.html#constrain-geoserver-worker-threads

> Constrain GeoServer Worker Threads
> 
> While the JVM provides memory management for most operations in Java applications, the memory used for rendering (in GeoServer’s case, responding to WMS GetMap requests) is not managed this way, so it is allocated in addition to the memory permitted by the JVM options above.
> 
> If GeoServer receives many concurrent requests, it may increase the memory usage significantly, so it is recommended to constrain the number of concurrent requests at the servlet container

Currently missing from https://github.com/georchestra/georchestra/blob/15.12/geoserver/webapp/src/docker/Dockerfile

**Comment by fvanderbiest** [comment]
See also https://wiki.eclipse.org/Jetty/Howto/High_Load

**Comment by fvanderbiest** [comment]
Currently in `/usr/local/jetty/etc/jetty.xml`:
```xml
    <!-- uncomment to change type of threadpool                                 
    <Arg name="threadpool"><New id="threadpool" class="org.eclipse.jetty.util.thread.QueuedThreadPool"/></Arg>                                                  
    -->                                                                         
    <Get name="ThreadPool">                                                     
      <Set name="minThreads" type="int"><Property name="threads.min" default="10"/></Set>                                                                       
      <Set name="maxThreads" type="int"><Property name="threads.max" default="200"/></Set>                                                                      
      <Set name="idleTimeout" type="int"><Property name="threads.timeout" default="60000"/></Set>                                                               
      <Set name="detailedDump">false</Set>                                      
    </Get>  
```

**Comment by fvanderbiest** [comment]
`threads.min` & `threads.max` come from `/usr/local/jetty/modules/server.mod`

**Comment by fvanderbiest** [comment]
```
$ cat /usr/local/jetty/modules/server.mod
#
# Base Server Module
#

[optional]
jvm
ext
resources

[lib]
lib/servlet-api-3.1.jar
lib/jetty-schemas-3.1.jar
lib/jetty-http-${jetty.version}.jar
lib/jetty-server-${jetty.version}.jar
lib/jetty-xml-${jetty.version}.jar
lib/jetty-util-${jetty.version}.jar
lib/jetty-io-${jetty.version}.jar

[xml]
etc/jetty.xml

[ini-template]
##
## Server Threading Configuration
##
# minimum number of threads
threads.min=10
# maximum number of threads
threads.max=200
# thread idle timeout in milliseconds
threads.timeout=60000
# buffer size for output
jetty.output.buffer.size=32768
# request header buffer size
jetty.request.header.size=8192
# response header buffer size
jetty.response.header.size=8192
# should jetty send the server version header?
jetty.send.server.version=true
# should jetty send the date header?
jetty.send.date.header=false
# What host to listen on (leave commented to listen on all interfaces)
#jetty.host=myhost.com
# Dump the state of the Jetty server, components, and webapps after startup
jetty.dump.start=false
# Dump the state of the Jetty server, before stop
jetty.dump.stop=false
# Enable delayed dispatch optimisation
jetty.delayDispatchUntilContent=false
```

**Comment by fvanderbiest** [comment]
With the following patch:
```
diff --git a/geoserver/webapp/src/docker/Dockerfile b/geoserver/webapp/src/docker/Dockerfile
index 744ba41..76b40ab 100644
--- a/geoserver/webapp/src/docker/Dockerfile
+++ b/geoserver/webapp/src/docker/Dockerfile
@@ -17,6 +17,8 @@ ADD . /
 
 VOLUME [ "/mnt/geoserver_datadir", "/mnt/geoserver_geodata", "/mnt/geoserver_tiles", "/tmp", "/run/jetty" ]
 
+RUN sed -i 's/threads.max=200/threads.max=50/g' /var/lib/jetty/start.d/server.ini
+
 CMD ["sh", "-c", "exec java -Djava.io.tmpdir=/tmp/jetty \
 -Dgeorchestra.datadir=/etc/georchestra \
 -Dgeofence.dir=/etc/georchestra/geoserver/geofence \
```

The max number of threads is indeed reduced:
```
# java -jar /usr/local/jetty/start.jar --list-config | grep threads
 threads.max = 50
 threads.min = 10
 threads.timeout = 60000
```

Opinions ?

**Comment by fvanderbiest** [comment]
> Opinions ?

@Vampouille is OK.

Fixed with 28b338f
